02.calendar/cal.rb

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env ruby
+
+require "date"
+require "optparse"
+
+# 引数を読み込む
+opt = OptionParser.new
+option = {}
+opt.on("-y") {|x| option[:y] = x}
+opt.on("-m") {|x| option[:m] = x}
+opt.parse!(ARGV)
+
+# 引数チェック
+if option[:y] && option[:m]
+    year = ARGV[0].to_i
+    month = ARGV[1].to_i
+    start_date = Date.new(year,month)
+    end_date = Date.new(year,month, -1)
+elsif option[:m]
+    date = DateTime.now
+    year = date.year
+    month = ARGV[0].to_i
+    start_date = Date.new(year,month)
+    end_date = Date.new(year,month, -1)
+elsif !(option[:y] && option[:m])
+    date = DateTime.now
+    year = date.year
+    month = date.month
+    start_date = Date.new(year,month)
+    end_date = Date.new(year,month, -1)
+end
+
+# 最後の日付を取得する
+end_number = end_date.day.to_i
+
+# 日付の表示のずれを調整するために0埋めの配列を作成する。
+zero_list = Array.new(start_date.wday) {|i| i = 0}
```

Comment:
この処理の場合、`i`は何の役にも立ってないのでそもそも省略可能です。
（必要なのはブロックの戻り値だけ）

```suggestion
zero_list = Array.new(start_date.wday) { 0 }
```

そもそもこの場合、ブロックを使う必要もありません。

```suggestion
zero_list = Array.new(start_date.wday, 0)
```

ただ、0に特別な意味を持たせるのはオレオレルールになってしまうのでコードの可読性を下げます。
「`start_date.wday`の数だけ`"   "`を繰り返し出力する」というようなロジックにすれば「特別な0」は登場させずに済むのではないでしょうか

Reply:
`start_date.wday`の数だけ`"   "`を繰り返し出力するロジックに変更いたしました。
おっしゃる通り、0埋めリストと日付リストをマージしても、
他の人が見た時に「0って何？」と思われちゃいますね。。。